third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/WMMA.cpp

```diff
@@ -168,51 +174,56 @@ LogicalResult convertDot(DotOp op, DotOpAdaptor adaptor,
   auto repB =
       wmmaLayout.getWMMARepForOperands(bTensorTy.getShape(), elemTy, kWidth, 1);
 
-  assert(repA[1] == repB[0]);
+  assert(repA[2] == repB[1]);
 
   Value loadedA = adaptor.getA();
   Value loadedB = adaptor.getB();
   Value loadedC = adaptor.getC();
-  auto numRepM = repA[0];
-  auto numRepN = repB[1];
-  auto numRepK = repA[1];
+  auto numRepM = repA[1];
+  auto numRepN = repB[2];
+  auto numRepK = repA[2];
+  auto numRepB = repA[0];
 
   ValueTable ha = getValuesFromDotOperandLayoutStruct(
-      rewriter, typeConverter, loadedA, numRepM, numRepK, kWidth,
+      rewriter, typeConverter, loadedA, numRepB, numRepM, numRepK, kWidth,
       aTensorTy.getElementType(), loc);
   ValueTable hb = getValuesFromDotOperandLayoutStruct(
-      rewriter, typeConverter, loadedB, numRepN, numRepK, kWidth,
+      rewriter, typeConverter, loadedB, numRepB, numRepN, numRepK, kWidth,
       aTensorTy.getElementType(), loc);
   auto dstElemTy = dTensorTy.getElementType();
   auto fc = unpackLLElements(loc, loadedC, rewriter);
 
   unsigned warpSize = triton::gpu::getWarpSize(wmmaLayout);
-  // TODO get rid of magic numbers
-  unsigned vgprElemWidth = 32;
+  constexpr unsigned vgprElemBitWidth = 32;
   unsigned paddedOutputElemSize =
-      vgprElemWidth / dstElemTy.getIntOrFloatBitWidth();
+      vgprElemBitWidth / dstElemTy.getIntOrFloatBitWidth();
   // compute number of output elements that each thread holds for one WMMA
   // instruction.
   auto elemsPerVec = mnkDim[0] * mnkDim[1] * paddedOutputElemSize / warpSize;
   auto dElemsToStorePerThread = mnkDim[0] * mnkDim[1] / warpSize;
   auto vecTy = vec_ty(dstElemTy, elemsPerVec);
-  for (int m = 0; m < numRepM; ++m) {
-    for (int n = 0; n < numRepN; ++n) {
-      Value acc = undef(vecTy);
-      for (unsigned v = 0; v < dElemsToStorePerThread; ++v) {
-        acc = insert_element(vecTy, acc,
-                             fc[m * numRepN * dElemsToStorePerThread +
+  for (int b = 0; b < numRepB; ++b) {
+    for (int m = 0; m < numRepM; ++m) {
+      for (int n = 0; n < numRepN; ++n) {
+        Value acc = undef(vecTy);
+        for (unsigned v = 0; v < dElemsToStorePerThread; ++v) {
+          acc =
+              insert_element(vecTy, acc,
+                             fc[b * numRepM * numRepN * dElemsToStorePerThread +
```

Comment:
I'm not a fan of an indexing that spans 3 lines. Can we use a local variable to make it clear?

Reply:
will change

FYI. Next step after this change is a refactoring, so this code will be different anyway: https://github.com/triton-lang/triton/pull/3600#discussion_r1558244091